### DISCUSSION TITLE
Layouts per screen

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#3688] [Q&A] [answered]
Is it possible to set layout stacks per screen (restrict each monitor to their own specific layout list).
If not, would it be hard to implement?

**Comment by elParaguayo** [comment, marked as answer]
Set layouts per group and restrict which groups can be on which screen.

http://docs.qtile.org/en/stable/manual/faq.html#how-can-i-get-my-groups-to-stick-to-screens

**Reply by ghost** [reply]
Thanks for the reply.
Yes. I've read about that.

Just to be sure: without wanting to restrict each screen's groups it isn't easily possible?

I've thought about something like (pseudocode):
```
layouts1 = [
    "max",
    "whatever",
    "floating",
]

layouts2 = [
    ...
]

screens = qtile.screen.info()
index = screens["index"]

if index == 0:
    layouts1.append(layouts1.pop(0))
    layout.set(layouts1[0])
elif index == 1:
    ...

```

**Reply by elParaguayo** [reply]
Can you explain your use case? I'm not sure I understand why you want to restrict layout per screen.

In qtile, layouts are not attached to screens and, if they were, your idea would mean that moving a group from one screen to the other could potentially alter the layout. That sounds odd to me but maybe I'm just misunderstanding what you're trying to do.

**Reply by ghost** [reply]
Just something I was thinking about and it's not really important to me.

I have two monitors, one horizontal, one vertical, so I thought it would be nice to switch between different layouts.

Example:
I like to use the `MonadTall` layout but on the vertical screen it doesn't make sense.
Therefore, replacing it with `MonadWide` on that screen would have been my idea.

I didn't give it too much thought, so yes, moving a group to another screen could be a problem.
I'll probably go with restricting the screen's groups then.

**Reply by elParaguayo** [reply]
I mean, it _is_ possible. I just haven't thought about the best way to do it!

**Reply by ghost** [reply]
I went with the method in your first response. Makes more sense anyway. Thank you!
